### PR TITLE
Added T-Flip-Flop

### PR DESCRIPTION
**Pull request by Rickgg**

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -197,6 +197,17 @@ test('SR-Flip-Flip : No Change', t => {
...
 })
 
+test('T-Flip-Flip: 1', t => {
+  const T = wires(1)
+  const qqbar = wires(2)
+  const clock = new Pulse(500, 1)
+  clock.switchOn()
+  const ff = new TFlipFlop(T, qqbar, clock)
+  const ioHandler = new StringIO(ff)
+  t.is(ioHandler.input('1'), '1')
```
Comment (mbad0la) [inline test.js]: ioHandler returns an empty string.

Also, as this is a T-Flip Flop, for a input `1` the output should be `0`.

File: test.js [hunk 2]
```diff
@@ -197,6 +197,17 @@ test('SR-Flip-Flip : No Change', t => {
   clock.switchOff()
 })
 
+test('T-Flip-Flip: 1', t => {
```
Comment (mbad0la) [inline test.js]: This test is failing!

File: Sequential/ff.js [hunk 3]
```diff
@@ -31,4 +31,20 @@ class DFlipFlop extends Hardware {
...
+class TFlipFlop extends Hardware {
+
+  constructor(t, qqbar, c) {
+    if (t.length != 1 || qqbar.length != 2) throw new Error('Invalid Connection/s')
+    super([t, [qqbar[0]]])
+    this.internalWiring = wires(4)
+    this.components.push(new AndGate([c], [qqbar[0]], [this.internalWiring[0]]))
+    this.components.push(new AndGate([c], [qqbar[1]], [this.internalWiring[1]]))
+    this.components.push(new AndGate(t, [this.internalWiring[0]], [this.internalWiring[2]]))
+    this.components.push(new AndGate(t, [this.internalWiring[1]], [this.internalWiring[3]]))
```
Comment (mbad0la) [inline Sequential/ff.js]: You could probably use `TriInpAndGate` component here instead of these four `AndGate` components.

File: test.js [hunk 4]
```diff
@@ -197,6 +197,17 @@ test('SR-Flip-Flip : No Change', t => {
...
+  const T = wires(1)
+  const qqbar = wires(2)
+  const clock = new Pulse(500, 1)
+  clock.switchOn()
+  const ff = new TFlipFlop(T, qqbar, clock)
+  const ioHandler = new StringIO(ff)
+  t.is(ioHandler.input('1'), '1')
+  clock.switchOff()
+})
+
```
Comment (mbad0la) [inline test.js]: As this library is in it's infancy, there might be some problems with how it handles building components from other components. Please do debug whether all the wires have appropriate values when a simulation is being run.

Also, as both qqbar[0] and qqbar[1] have an initial value of `undefined`, maybe that's why your component is not able to toggle it.

File: test.js (hunk 2 above)
Comment (Rickgg) [inline test.js]: Mmm it worked for me...

**Comment by Rickgg** [comment]
I will check it again

**Pull request closed without merging**